src/components/TextInput/TextInput.tsx

```diff
@@ -76,6 +91,12 @@ export const TextInput = React.forwardRef<HTMLSpanElement, TextInputProps>(funct
         }
     }, [multiline, inputValue, hasVerticalScrollbar]);
 
+    React.useLayoutEffect(() => {
+        const labelEl = labelRef.current;
+        const width = labelEl?.offsetWidth || 0;
+        setLabelWidth(width);
+    }, [labelSize, label, size, multiline]);
```

Comment:
Can we get correct size in `useElementSize`? [ResizeObserverEntry](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry) may have [borderBoxSize](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry/borderBoxSize), but if it doesn't, it has a `target`, we can get `offsetWidth` from it.